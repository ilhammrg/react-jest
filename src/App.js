import React, { useEffect, useState } from "react";
import Child from './components/child/Child';

const App = () => {
  const [user, setUser] = useState("");

  async function getData() {
    const response = await fetch(
      "https://jsonplaceholder.typicode.com/users/1"
    );
    const data = await response.json();
    setUser(data.name);
  }

  useEffect(() => {
    getData();
  }, []);

  return (
    <>
      <h1>Hello world!</h1>
      <Child />
      <p>{user}</p>
    </>
  );
};

export default App;
