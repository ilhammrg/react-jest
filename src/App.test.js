import React from 'react';
import { shallow } from 'enzyme';

// component to test
import App from './App';

describe('App component test', () => {
  test('Should render App component', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});