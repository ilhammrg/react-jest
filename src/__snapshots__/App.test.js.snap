// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test Should render App component 1`] = `
<Fragment>
  <h1>
    Hello world!
  </h1>
  <Child />
  <p />
</Fragment>
`;
